EIPS/eip-beacon_chain_push_withdrawals.md

```diff
@@ -0,0 +1,121 @@
+| `WITHDRAWAL_TX_TYPE`          | `0x3`                                          | byte
+
+Beginning with the execution timestamp `FORK_TIMESTAMP`, execution clients **MUST** introduce the following extensions to transaction processing and block validation:
+
+### New transaction type
+
+Define a new [EIP-2718](./eip-2718.md) transaction type with `TransactionType` `WITHDRAWAL_TX_TYPE`.
+
+The `TransactionPayload` is an SSZ-encoded container given by the following schema:
+
+```python
+class WithdrawalTransaction(Container):
+    address: ExecutionAddress
+    amount: Gwei
+```
+
+where `ExecutionAddress` is an alias for a `Bytes20` SSZ type and `Gwei` is an alias for a `uint64` SSZ type.
+Refer to the [SSZ specs](https://github.com/ethereum/consensus-specs/blob/master/ssz/simple-serialize.md) for further details on layout and encoding.
+
+### Block validity
+
+If a block contains *any* transactions with `WITHDRAWAL_TX_TYPE` type, they **MUST** come before **ALL** other transactions in the block.
+
+If the execution client receives a block where this is not the case, it **MUST** consider the block invalid.
+
+### Transaction processing
+
+When processing a transaction with `WITHDRAWAL_TX_TYPE` type, the implementation should make a balance transfer for `amount` Gwei to the `address` specified by the `WithdrawalTransaction`.
+There is no source for this balance transfer, much like the coinbase transfer.
+
+This balance transfer is unconditional and **MUST** not fail.
+
+This transaction type has no associated gas costs.
+
+TODO: add logs?
+
+## Rationale
+
+### Push vs pull
+
+cheaper for validators
+
+happens automatically
+
+### Why a new transaction type? And why no gas?
+
+Special semantics
+
+Firewall off generic EVM excution from this type of processing to simplify testing, security review.
+
+### Why no gas costs for new transaction type?
+
+The maximum number of this transaction type that can reach the execution layer at a given time is bounded (enforced by the consensus layer) and this limit is kept small so that
+any gas required is negligible in the context of the broader block gas limit.
+
+### Why only balance updates? No general EVM execution?
+
+more general processing introduces the risk of failures, which complicates accounting on the beacon chain
+
+generic execution doesn't pull its weight given this increase in complexity
+
+### Why block validations?
+
+ensure only those receipts genuinely made on the beacon chain are passed to the EVM
+
+provide efficient means for beacon chain to check this
+
+### SSZ vs RLP, and their boundary
+
+< explain where we put the translation boundary from SSZ to RLP >
+
+point out that the transaction hash is hash of RLP of SSZ encoding, not the hash tree root of the SSZ
+
+## Backwards Compatibility
+
+No issues.
+
+## Reference Implementation
+
+A draft PR containing a prototype implementation in Geth can be found here:
+
+https://github.com/ethereum/go-ethereum/pull/24468
+
```

Comment:
```suggestion
```
Either inline a reference implementation, or include a standalone reference implementation in `../assets/eip-4863/*`, or leave this section out.  Linking to a pull request to an entire execution client does not qualify as a "reference implementation" for an EIP, that is just called a full production client.

Reply:
i think a link to a PR here is a great way to showcase the relative complexity of this change (esp. if you are familiar w/ Geth implementation).
 
i can remove it if you think it blocks this EIP but i'll just link it everywhere else

Reply:
done!